File: use_after_free.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

int main()
{
    char *hello = malloc(10 * sizeof(*hello));

    strcpy(hello, "Hello!");
    printf("hello=%s\n", hello);

    free(hello);

    // NOTE: use-after-free...
    printf("hello=%s\n", hello);
}

File: use_after_free.rs



#[allow(unused_assignments)]
fn main()
{
    let mut hello = String::new();

    hello = String::from("Hello!");
    println!("hello={}", hello);

    // NOTE: It is not possible to free memory
    // manually within the safe Rust environment

    println!("hello={}", hello);
}
